BlueprintUICommonControls/Sources/AttributedLabel+Environment.swift

```diff
@@ -17,14 +17,19 @@ public protocol URLHandler {
     func onTap(url: URL)
 }
 
-struct DefaultURLHandler: URLHandler {
-    func onTap(url: URL) {
+class NullURLHandler: URLHandler {
+    func onTap(url: URL) {}
+}
+
+class DefaultURLHandler: NullURLHandler {
+    @available(iOSApplicationExtension, unavailable)
+    override func onTap(url: URL) {
         UIApplication.shared.open(url, options: [:], completionHandler: nil)
     }
 }
 
 public struct URLHandlerEnvironmentKey: EnvironmentKey {
-    public static let defaultValue: URLHandler = DefaultURLHandler()
+    public static var defaultValue: URLHandler = DefaultURLHandler()
```

Comment:
In a future version, we could change the default to be a no-op

Reply:
Hmm, I don't feel great about having a mutable global like this — I think if you need to have clickable links in an app extension context, it should be on you to accomplish that through normal environment means, i.e. adding an  `.adaptedEnvironment()` modifier.

Reply:
Ah shit, I missed this was globally mutable like this – it does seem easy to unintentionally break. I agree with Andrew's take here.

Reply:
I spoke to Adam on Slack and suggested we add a runtime check here (by [inspecting the main bundle](https://stackoverflow.com/a/29322061)), and do something like:

```swift
public static let defaultValue: URLHandler = runningInAppContext ? NullURLHandler() ? DefaultURLHandler()
```